src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoParser.cs

```diff
@@ -3,224 +3,232 @@
 using System.IO;
 using System.Linq;
 using System.Text;
+using System.Threading.Tasks;
 
-namespace OrchardCore.Localization.PortableObject
+namespace OrchardCore.Localization.PortableObject;
+
+/// <summary>
+/// Represents a parser for portable objects.
+/// </summary>
+public class PoParser
 {
+    private static readonly Dictionary<char, char> _escapeTranslations = new()
```

Comment:
Perfect candidate for FrozenDictionary.
Or just remove this dictionary and use a `switch` the compiler is clever enough to use a dictionary of ifs based on the number of items for best performance, so we don't have to think about it too much.

Reply:
I will create another PR for that because it might need extra testing and benchmarking